### PR TITLE
feat: update readme

### PR DESCRIPTION
**Pull request by stevelouiscooper**
## Description

Changes to README to cover 

- Technology overview.
- Environment setup.
- Component development.
- Component installation.
- Component release.
- Figma integrations.
- Linking FE applications.
- Key files.

I have intentionally left theming out of the docs as it is still WIP.

### Commitment to review

- If you already have bit.dev on your machine reading the documentation should be enough, roughly a 5 minute commitment.
- If you do not have bit.dev on your machine I would encourage you to follow the documentation and set up an environment, which can take up to 30 minutes depending on how far you decide to follow.

### Review instructions

Give the doc a read and if you have the time or inclination set up a local environment using the instructions provided.

Probably best to view the doc [here](https://github.com/KoodooHQ/K2/tree/feat/update-readme).

### Acceptance criteria

  - [ ] Docs make sense
  - [ ] Docs are useful
  - [ ] LGTM :+1:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,110 @@
...
+
+``` 
+
+
+### Publishing your components :tada:
+
+- Once you're happy with the development and testing of your components and are ready to have them reviewed, be sure to run the following commands as a plenary exercise so you're ready to write a great pull request.
+- `bit status` to see if any issues are flagged by Bit.Dev based on recent changes. If it suggests you have items to resolve, run the relevant commands it suggests and run `bit status` again to ensure you're all clear. 
+- `bit test --all` to run tests over all components and ensure they're passing.
+- `bit tag --all` to tag all components with differences so they will be version bumped on successful merge with `main`.
```
Comment (Tommoore96) [inline README.md:86]: Mention using the `--soft` flag here, it's the github actions that then use the soft tags to hard tag it.

**Review by jose-verissimo — APPROVED** [review]
LGTM 💯 
Was able to read it clearly and get bit.dev running locally with K2's components 👌

File: README.md [hunk 2]
```diff
@@ -1,33 +1,110 @@
...
+## Good to know :exclamation:
+
+- **workspace.jsonc**
+
+This is the main configuration file of your bit workspace. Here you can modify the workspace name and icon as well as default directory and scope. It is where dependencies are found when you install anything. It is also where you register aspects, bit extensions as well as apply the environments for your components. This workspace has been setup so that all components use the React env. However you can create other components and apply other envs to them such as node, html, angular and aspect envs.
+
+- **.bitmap**
 
-Ignoring any files from version control
+This is an auto-generated file and includes the mapping of your components. There is one component included here. In order to remove this component you can run the following command.
```
Comment (Tommoore96) [inline README.md:110]: Which following command 😰 

Tbh I think the most important thing to note here is that this file shouldn't be edited manually for *any* reason.  If you want to add, remove, bump any component it should be done from the CLI instead.

File: README.md [hunk 3]
```diff
@@ -1,33 +1,110 @@
...
+- Create a new react component by running `bit create react-component ui/{component-name}`
+- This will create some template files that will be used to customise your component.
+  - `{component-name}.spec.tsx` file for testing your component.
+  - `{component-name}.composition.tsx` file for showcasing different configurations of your component.
+  - `{component-name}.docs.tsx` file for presenting a mock implementation of your component and presenting a live code example for viewers to interact with. This is also the file that can display Figma designs.
+  - `{component-name}.tsx` file for creating your component.
+- Run `bit compile` to compile the components (necessary when adding new components).
+- Run `bit start` to start the dev server, you should see your component and any edits you make should hot reload the application views :zap:
+
+### Showing Figma designs
```
Comment (Tommoore96) [inline README.md:54]: MINOR: I dunno if it would help to mark this as "(optional)" or something

Comment (stevelouiscooper) [reply README.md:54]: Showing Figma designs?

File: README.md (hunk 2 above)
Comment (stevelouiscooper) [reply README.md]: Good shout, added that.

File: README.md [hunk 4]
```diff
@@ -1,33 +1,110 @@
...
+
+### Initialise your Bit.Dev workspace
+
+- Use your current terminal window (or a new one) to navigate to the root directory of the project and run `bit status`
+  - This command should identify a workspace for you, but if it does not we can initialise it ourselves using `bit init` then running `bit status` again.
+- The terminal will likely suggest some commands to run such as:
+  - `bit install` to install development workspace dependencies.
+  - `bit import` to import components into your current workspace.
+  - `bit compile` to transform the source code into executable code that can run in multiple browsers.
+- Follow the prompts given to you when you run `bit status` until there are no more suggested commands to run. At this point you will have successfully initialised your Bit.Dev workspace! :star2:
```
Comment (peelyKoodoo) [inline README.md:32]: For me this was not a case of a suggestion, I got lots of errors until I had run all of these commands, and `bit status` did not suggest the compile command. To me I think it was `bit init && bit import && bit compile && bit install`, and then `bit start` worked.

**Review by peelyKoodoo — APPROVED** [review]
I'm happy if you're happy, it's a great step forward and we can always add more!

**Pull request merged**